src/platform.ts

```diff
@@ -0,0 +1,98 @@
+import os from 'node:os'
+
+enum Platform {
+  Linux,
+  MacOS,
+  Windows
+}
+
+enum Architecture {
+  arm64,
+  x64
+}
```

Comment:
I generally prefer `as const` objects or arrays instead of enums - enums are one of the few TypeScript features which are transpiled into entirely different runtime code (whereas most TypeScript type-related syntax is simply erased).